framework/src/parser/Parser.C

```diff
@@ -139,425 +238,320 @@ Parser::getFileName(bool stripLeadingPath) const
-                    "(.*)"   // the variable override that applies to the MultiApp
-                    )
-            .FullMatch(all_vars[i], &multi_app, &app_num, &variable) &&
-        fe_problem->hasMultiApp(multi_app) && // Make sure the MultiApp exists
-        // Finally make sure the number is in range (if provided)
-        static_cast<unsigned int>(app_num) < fe_problem->getMultiApp(multi_app)->numGlobalApps())
-
-      // delete the current item by copying the last item to this position and decrementing the
-      // vector end position
-      all_vars[i] = all_vars[last--];
-    else if (pcrecpp::RE("(.*?)" // Same as above without the MultiApp number
-                         ":"
-                         "(.*)")
-                 .FullMatch(all_vars[i], &multi_app, &variable) &&
-             fe_problem->hasMultiApp(
-                 multi_app)) // Make sure the MultiApp exists but no need to check numbers
-
-      // delete (see comment above)
-      all_vars[i] = all_vars[last--];
-
-    // TODO: check to see if globals are unused
-    else if (all_vars[i].find(":") == 0)
-      all_vars[i] = all_vars[last--];
-
-    else
-      // only increment if we didn't "delete", otherwise we'll need to revisit the current index
-      // due
-      // to copy
-      ++i;
-  }
-
-  mooseAssert(last + 1 >= 0, "index \"last\" is negative");
-
-  // Remove the deleted items
-  all_vars.resize(last + 1);
-
-  std::set<std::string> difference;
-
-  std::sort(all_vars.begin(), all_vars.end());
+  // Save the filename
+  char abspath[PATH_MAX + 1];
+  realpath(input_filename.c_str(), abspath);
+  _input_filename = std::string(abspath);
 
-  // Remove flags, they aren't "input" parameters
-  all_vars.erase(std::remove_if(all_vars.begin(),
-                                all_vars.end(),
-                                [](const std::string & s) { return s.length() && s[0] == '-'; }),
-                 all_vars.end());
+  // vector for initializing active blocks
+  std::vector<std::string> all = {"__all__"};
 
-  std::set_difference(all_vars.begin(),
-                      all_vars.end(),
-                      _extracted_vars.begin(),
-                      _extracted_vars.end(),
-                      std::inserter(difference, difference.end()));
+  MooseUtils::checkFileReadable(_input_filename, true);
 
-  // Remove un-parsed parameters that were located in an inactive sections
-  for (const auto & inactive_string : _inactive_strings)
-    for (std::set<std::string>::iterator j = difference.begin(); j != difference.end();
-         /*no increment*/)
-    {
-      std::set<std::string>::iterator curr = j++;
-      if (curr->find(inactive_string) != std::string::npos)
-        difference.erase(curr);
-    }
-
-  std::set<std::string> requested_vars = _getpot_file.get_requested_variables();
+  std::ifstream f(_input_filename);
+  std::stringstream ss;
+  ss << f.rdbuf();
+  std::string input = ss.str();
```

Comment:
This is a weird idiom. I know what you are doing, but it seems a little nonstandard. I googled "ifstream read in whole file" and came across this: https://stackoverflow.com/questions/2602013/read-whole-ascii-file-into-c-stdstring, it seems there are other perhaps better ways of reading in the whole buffer.